resourseleak/TestTryWithResource.java

```diff
@@ -0,0 +1,26 @@
+package resourceleak;
+
+import com.amazonaws.services.s3.model.S3Object;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+
+public class TestTryWithResource {
+
+    public String testTryWithResource(String bucket, String key) {
+
+        LOG.info("Loading " + bucket + ":" + key + " to string");
+        final GetObjectRequest request = new GetObjectRequest(bucket, key);
+        try (final S3Object object = s3Client.getObject(request);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code retrieves and decodes the contents of an S3 object to a String.
This can be done by calling [getObjectAsString](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3.html#getObjectAsString-java.lang.String-java.lang.String-). Using this function simplifies the code and eliminates the need to close the S3 object once it is read.